xcube_cci/cciodp.py

```diff
@@ -876,8 +888,11 @@ async def _get_data_chunk(self, session, request: Dict, dim_indexes: Tuple) -> O
         dataset = await self._get_opendap_dataset(session, opendap_url)
         if not dataset:
             return None
+        await self._ensure_all_info_in_data_sources(session, [request.get('drsId')])
+        data_type = self._data_sources[request['drsId']].get('variable_infos', {}).\
+            get(var_name, {}).get('data_type')
```

Comment:
PEP-8:
```suggestion
        data_type = self._data_sources[request['drsId']].get('variable_infos', {}) \
            .get(var_name, {}).get('data_type')
```

Reply:
The upcasting happens here:
https://github.com/dcs4cop/xcube-cci/blob/f0709567741f76e660d74f4aa23955204f77d277/xcube_cci/cciodp.py#L895

Reply:
> The upcasting happens here:
> https://github.com/dcs4cop/xcube-cci/blob/f0709567741f76e660d74f4aa23955204f77d277/xcube_cci/cciodp.py#L895

Numpy provides a cast operation which should be used instead. Also copying of array data should be avoided so we don't use more memory than required: `data.astype(dtype, casting='safe', copy=False)`